### PR TITLE
[WIP] Update various CI dependencies prior to a new release

### PR DESCRIPTION
**Pull request by kwilczynski**
#### What type of PR is this?

/kind ci
/kind dependency-change
/assign kwilczynski

#### What this PR does / why we need it:

Update various CI dependencies prior to a new release. This change also closes several dependency changes that had been open for a while.

Related:

- https://github.com/cri-o/cri-o/pull/7512
- https://github.com/cri-o/cri-o/pull/7519
- https://github.com/cri-o/cri-o/pull/7520
- https://github.com/cri-o/cri-o/pull/7521
- [golangci/golangci-lint: v1.55.2](https://github.com/golangci/golangci-lint/releases/tag/v1.55.2)

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

None

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Review by saschagrunert — APPROVED** [review]
Thanks!

File: .github/workflows/osff.yml [hunk 1]
```diff
@@ -28,12 +28,12 @@ jobs:
...
     steps:
       - name: "Checkout code"
-        uses: actions/checkout@v4 # v3.1.0
+        uses: actions/checkout@v4
         with:
           persist-credentials: false
 
       - name: "Run analysis"
-        uses: ossf/scorecard-action@0864cf19026789058feabb7e87baa5f140aac736 # v2.3.1
+        uses: ossf/scorecard-action@v2
```
Comment (sohankunkerkar) [inline .github/workflows/osff.yml:36]: This change fails --> https://github.com/cri-o/cri-o/actions/runs/7113368129/job/19365168888

Comment (kwilczynski) [reply .github/workflows/osff.yml:36]: **Also, to add:**

This Pull Request was merged before I could fix some of the issues, as between me making it as "WIP", it was approved, and the bot was kind enough to merge it.

There is now https://github.com/cri-o/cri-o/pull/7560, which aims to fix some of these.

Comment (kwilczynski) [reply .github/workflows/osff.yml:36]: Done.

Changed in https://github.com/cri-o/cri-o/pull/7560/commits/950f5adcf95e4b4fa599321bd6271e0e64e75fab.

**Pull request merged**